src/wp-includes/class-wp-theme-json.php

```diff
@@ -1925,6 +1925,53 @@ public function get_styles_block_nodes() {
 		return static::get_block_nodes( $this->theme_json );
 	}
 
+	/**
+	 * Returns a filtered declarations array if there is a separator block with only a background
+	 * style defined in theme.json by adding a color attribute to reflect the changes in the front.
+	 *
+	 * @since 6.1.1
+	 *
+	 * @param array $declarations List of declarations.
+	 *
+	 * @return array $declarations List of declarations filtered.
+	 */
+	private static function update_separator_declarations( $declarations ) {
+		$background_matches = array_values(
+			array_filter(
+				$declarations,
+				function( $declaration ) {
+					return 'background-color' === $declaration['name'];
+				}
+			)
+		);
+		if ( ! empty( $background_matches && isset( $background_matches[0]['value'] ) ) ) {
+			$border_color_matches = array_values(
+				array_filter(
+					$declarations,
+					function( $declaration ) {
+						return 'border-color' === $declaration['name'];
+					}
+				)
+			);
+			$text_color_matches   = array_values(
+				array_filter(
+					$declarations,
+					function( $declaration ) {
```

Comment:
A tiny micro-optimization is to define the callback as `static`:

```suggestion
					static function( $declaration ) {
```